content/raw/chat/username_avatar.md

```diff
@@ -0,0 +1,132 @@
+- Display names MUST NOT end in "_eth" or ".eth" or "-eth"
+- Profiles containing 3 word names as their display name that does not deterministically correspond to their chatkey MUST display a warning icon to indicate that the username is not deterministic
+- A user MUST be able to set a local username for any profile other than their own
+- A local username MUST be visible only to the user who set it
+- If profile has a local set username set then it MUST show the original display name (e.g if the local username is 'intimidating guy' and the original display name is 'Jotaro' then `intimidating guy (Jotaro)`)
+- If a profile has a registered ENS username, then it MUST end in .eth
+- If a profile has a registered ENS stateofus.eth username, then it MUST end in stateofus.eth
+- If a profile has a registered ENS username, then it MAY NOT display the chat key next to the username
+- If a profile does not have a registered ENS username, then it MUST display the chat key next to the username
+
+**Profile avatar**
+
+- If a profile does not have a profile photo set it MUST use a 2 letter avatar
+- A 2 letter avatar MUST be the first two letters of the display name
+
+- Users MUST be able to set a profile photo for their profile
+- Profile photos MUST be visible to all users
+
+- A identicon ring MUST be composed of 5 color segments
+- A identicon ring segment MAY be one of 32 distinctive colors
+- A identicon ring segment MAY be of different length
+- A identicon ring segments total length MUST BE exactly 25px
+- unless a profile is a mutual contact, it MUST show the identicon ring around the profile avatar
+
+**EmojiHash**
+
+- An Emoji Hash MUST be a sequence of 12 emojis determined by the profile's chat key
+- A profile detailed view MUST show that profiles emoji hash
+
+**Profile actions**
+
+- If a profile is not a contact AND is not marked then
+    - The user MUST be able to mark that profile as untrustworthy
+    - The user MUST be able to block that profile
+- If a profile is not a contact AND is marked as untrusthworthy then
+    - The user MUST be able to remove the untrustworthy mark
+    - The user MUST be able to block that profile
+- if a profile is a contact AND is not marked then
+    - The user MUST be able to remove the untrustworthy mark
+    - The user MUST NOT be able to block or unblock that profile
+- if a profile is a contact AND is marked as untrusthworthy then
+    - The user MUST be able to remove the untrustworthy mark
+    - The user MUST NOT be able to block or unblock that profile
+- if a profile is a contact AND has the identity verified then
+    - The user MUST NOT be able to mark that profile as untrustworthy or trusthworthy
+    - The user MUST NOT be able to block or unblock that profile
+
+**Usernames in Communities**
+
+- A community MUST NOT allow duplicate display names
+
+## Notes
+
+- This spec implies that identicons & 3 word usernames features are to be removed from the app.
```

Comment:
Maybe worth to add that they should be still supported for backwards compatibility (showing the 3 word names for people still on an old version)